neuro_san/http_sidecar/http_sidecar.py

```diff
@@ -132,11 +136,34 @@ def update_agents(self, metadata: Dict[str, Any]):
         with self.lock:
             # We assume all agents from "agents" list are enabled:
             for agent_name in agents:
-                self.allowed_agents[agent_name] = True
+                if self.allowed_agents.get(agent_name, None) is None:
+                    self.add_agent(agent_name)
             # All other agents are disabled:
             for agent_name, _ in self.allowed_agents.items():
                 if agent_name not in agents:
-                    self.allowed_agents[agent_name] = False
+                    self.remove_agent(agent_name)
+
+    def add_agent(self, agent_name: str):
+        """
+        Add agent to the map of known agents
+        :param agent_name: name of an agent
+        """
+        agent_tool_factory_provider: SingleAgentToolFactoryProvider =\
+            ServiceToolFactoryProvider.get_instance().get_agent_tool_factory_provider(agent_name)
+        # Convert back to a single string as required by constructor
+        request_metadata_str: str = " ".join(self.forwarded_request_metadata)
+        agent_server_logging: AgentServerLogging =\
+            AgentServerLogging(self.server_name_for_logs, request_metadata_str)
+        agent_service: AsyncAgentService =\
+            AsyncAgentService(self.logger, None, agent_name, agent_tool_factory_provider, agent_server_logging)
+        self.allowed_agents[agent_name] = agent_service
```

Comment:
Construct AsyncAgentService instance for an agent if we need one,
then add it to our internal table.